generators/cmd/predefined/profiles.go

```
@@ -61,6 +64,32 @@ func getProfile() (*profile, error) {
 	return loadedProfile, nil
 }
 
+func getProfileFromExternalCommand(command string) (*profile, error) {
+	args, err := shellwords.Parse(command)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(args) < 1 {
+		return nil, nil
```

just a comment: actually, I'm not sure if keeping silent here is better. I guess even if the user specifies the `profile-command` option but there is nothing then that would be the unexpected thing, isn't it? Possibly would it be better to raise an error?

---

@moznion 
Thanks for your comment!
If the command is unspecified, `getProfileFromExternalCommand()` is not called.
Therefore, this condition is probably only caused by a bug in `shellwords.Parse()` and is unlikely to occur in the first place.

If the function returns null here, a null reference error will occur at the caller, and if it does not return null (and is left silent), an error will occur on the reference to `args[1:]`, so the difference is whether the error occurs inside or outside the function.
Neither is very well behaved, so explicitly returning an error would be a more beautiful implementation.

コメントありがとうございます！
コマンドが未指定の場合は`getProfileFromExternalCommand()`が呼ばれませんので、この条件が発生するのは恐らく`shellwords.Parse()`のバグ位で、そもそも発生することがほぼ無いとは思います。

関数がここでnullを返すと呼び出し元でnull参照のエラーが発生し、nullを返さない(silentにしておく)と`args[1:]`の参照でエラーが発生するので、エラーが関数の中で起こるか関数の外で起こるかの違いになります。
いずれもあまり行儀は良くないので、明示的にエラーを返す方が実装としては美しそうですね。

---

just a comment: I think this error rarely occurs, so I will leave it in this way.